text/0796-ember-data-deprecate-rsvp.md

```diff
@@ -0,0 +1,76 @@
+---
+Stage: Proposed
+Start Date: 2022-02-20
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): ember-data 
+RFC PR: https://github.com/emberjs/rfcs/pull/796
+---
+
+# Deprecate `RSVP.Promise` for native Promise
+
+## Summary
+
+All methods currently returning an `RSVP.Promise` will return a native Promise. In addition, all documentation will be updated accordingly.
+
+## Motivation
+
+With the removal of IE11 from our build infrastructure and browser requirements, we can now safely remove `RSVP.Promise`. From the [docs](https://github.com/tildeio/rsvp.js/)
+
+- Specifically, it is a tiny implementation of Promises/A+. It works in node and the browser (IE9+, all the popular evergreen ones).
+
+RSVP was Ember's polyfill since an early v1.0.0 version during a time when native Promises didn't exist in any browser.  In addition, Promises have been supported since Node 0.12.
+
+By removing `RSVP.Promise` in favor of native Promises, we can drop an unnecessary dependency for both client side and server side fetching of data.
+
+According to [bundlephobia](https://bundlephobia.com/package/rsvp@4.8.5), this would allow us to remove a significant chunk of dependency weight.
+
+## Detailed design
+
+Two steps will be required to fulfill deprecating `RSVP.Promise`.
+
+First, we will issue a deprecation to async callbacks that might be relying on a convenient feature of `RSVP.Promise`.  Namely, an `RSVP.Promise` may still be hanging by the time the underlying model instance or store has been destroyed.  This will help users surface instances where their test suite or code is dealing with dangling promises.  After the removal of this deprecation, this will throw an Error.
+
+- `id: ember-data:rsvp-promise-hanging`
+
+Second, we will introduce a feature flag replacing all instances of `RSVP.Promise` with native Promises.
+
+The implementation follows the standard feature-flagging process in Ember and Ember Data. The final implementation will look something like:
+
+```js
+  ajax(options) {
+    if (DS__NATIVE_PROMISE) {
+      return new Promise((resolve, reject) => {
+        ...
+      });
+    } else {
+      return new RSVPPromise((resolve, reject) => {
+        ...
+      });
+    }
+  },
+```
+
+## How we teach this
+
+We do not believe this requires any update to the Ember curriculum. API documentation may be needed to remove traces of `RSVP.Promise`.
```

Comment:
This needs to include the information that goes into the deprecation guide.
You will also need to document the optional feature, especially if you want to flip the availability down the line, right?